extras/gizmo/tri.js

```diff
@@ -0,0 +1,73 @@
+import {
+    Vec3
+} from 'playcanvas';
+
+const e1 = new Vec3();
+const e2 = new Vec3();
+const h = new Vec3();
+const s = new Vec3();
+const q = new Vec3();
+
+// constants
+const EPSILON = 1e-6;
+
+class Tri {
+    v0 = new Vec3();
+
+    v1 = new Vec3();
+
+    v2 = new Vec3();
+
+    constructor(v0 = new Vec3(), v1 = new Vec3(), v2 = new Vec3()) {
```

Comment:
I'd keep the constructor simple:
- do not create it from an array (you're not using it)
- do not create default `new Vec3` - those are unused when created from the array
- just expect 3 Vec3's being specified. Creating a triangle with three zero vertices is not very useful.